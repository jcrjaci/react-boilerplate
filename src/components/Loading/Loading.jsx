import React from 'react';
import PropTypes from 'prop-types';

import './Loading.scss';

const Loading = ({ className }) => (
  <div className={`loading ${className}`}>
    <div className="spinner">
      <div className="rect1" />
      <div className="rect2" />
      <div className="rect3" />
      <div className="rect4" />
      <div className="rect5" />
    </div>
  </div>
);

Loading.propTypes = {
  className: PropTypes.string,
};

Loading.defaultProps = {
  className: '',
};

export default Loading;
