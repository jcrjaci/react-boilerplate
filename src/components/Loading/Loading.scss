// @import "../../styles/variables";

.loading {
  align-items: center;
  display: flex;
  height: 40px;
  justify-content: center;
  transition: height, .5s linear;

  @keyframes sk-stretchdelay {

    0%,
    40%,
    100% { transform: scaleY(.4) }  
    
    20% { transform: scaleY(1.0) }
  }
  
  @keyframes sk-stretchdelay {
    
    0%,
    40%,
    100% { 
      transform: scaleY(.4);
    }
    
    20% { 
      transform: scaleY(1.0);
    }
  }
    
  .spinner {
    font-size: 10px;
    height: 40px;
    text-align: center;
    width: 50px;
  }

  .spinner div {
    animation: sk-stretchdelay 1.2s infinite ease-in-out;
    background-color: #000;//$color-input;
    display: inline-block;
    height: 100%;
    margin: 0 1px;
    width: 6px;
  }

  .spinner .rect2 {
    animation-delay: -1.1s;        
  }

  .spinner .rect3 {
    animation-delay: -1.0s;
  }

  .spinner .rect4 {
    animation-delay: -.9s;
  }

  .spinner .rect5 {
    animation-delay: -.8s;
  }
    
  &.hide {
    height: 0;
    transition: height, .5s linear;

    .spinner {
      height: 0;
    }
  }
}
