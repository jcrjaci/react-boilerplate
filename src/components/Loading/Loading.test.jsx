import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Loading from './Loading';

configure({ adapter: new Adapter() });

describe('<Loading />', () => {
  const wrapper = shallow(<Loading />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    wrapper.setProps({ className: 'someClass' });
    expect(wrapper.find('.loading').hasClass('someClass')).toBeTruthy();
  });
});
