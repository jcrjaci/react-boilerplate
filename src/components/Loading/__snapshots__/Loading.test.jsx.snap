// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render correctly 1`] = `
<div
  className="loading "
>
  <div
    className="spinner"
  >
    <div
      className="rect1"
    />
    <div
      className="rect2"
    />
    <div
      className="rect3"
    />
    <div
      className="rect4"
    />
    <div
      className="rect5"
    />
  </div>
</div>
`;
